pdns/misc.hh

```diff
@@ -112,7 +112,8 @@ stringtok (Container &container, string const &in,
 
 template<typename T> bool rfc1982LessThan(T a, T b)
 {
-  return ((signed)(a - b)) < 0;
+  typedef typename std::make_signed<T>::type signed_t;
```

Comment:
It's not clear to me what `std::make_signed` does if we ever pass a signed type by mistake, perhaps we could add a `static_assert()` using `std::is_unsigned()` ?